application/common/core.cpp

```diff
@@ -71,11 +71,12 @@ void loop()
     /* Check for events */
     while (SDL_PollEvent(&event))
     {
-        if (event.type == SDL_QUIT)
+        if (event.type == SDL_QUIT ||
+            event.type == SDL_APP_TERMINATING)
```

Comment:
I thought about that too, but it wouldn't really help much. Even if it pops up in another thread (on some devices), you still have to wait for the tick to finish or force an exit.